File: service/board-service/src/main/java/jeonghyeon/msa/board/kafka/handle/EventHandler.java
package jeonghyeon.msa.board.kafka.handle;

import jeonghyeon.msa.common.event.Event;
import jeonghyeon.msa.common.event.EventPayload;

public interface EventHandler <T extends EventPayload> {
    void handle(Event<T> event);
    boolean supports(Event<T> event);
}

File: service/memo-service/src/main/kotlin/jeonghyeon/msa/memo/service/event/handler/EventHandler.kt
package jeonghyeon.msa.memo.service.event.handler

import jeonghyeon.msa.common.event.Event
import jeonghyeon.msa.common.event.EventPayload

interface EventHandler<T : EventPayload> {
    fun handle(event: Event<T>)
    fun supports(event: Event<T>): Boolean
}